### PR TITLE
Ho 20 creating listing route

### PR DESCRIPTION
**Pull request by SyntheticDBX**
Added changes requested from old pull request for the address object in the listing schema. 

Also added route for listings and tested in postman to confirm they are working. Data can be see in the database.

### CODE REVIEW

File: backend/controllers/listingsController.js [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+const getAllListings = async (req, res) => {
+  const listings = await Listing.find({}).sort({createdAt: -1});
+  res.status(200).json(listings);
+}
+
+const createListing = async (req, res) => {
+  const {organisationName, title, description, address, neededByDate} = req.body;
+  let emptyFields = [];
+  if (!organisationName){
```
Comment (Sabirah42) [inline backend/controllers/listingsController.js:13]: Is an if/else statement necessary? Could this be done with a map?

File: backend/models/listingModel.js [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+  address: {
+    firstLine: {
+      type: String,
+      required: true
+    },
+    city: {
+      type: String,
+      required: true
+    },
+    postcode: {
```
Comment (Sabirah42) [inline backend/models/listingModel.js:27]: Needs some validation added - length and postcode pattern (using regex?). Important as we're going to be using this field for mapping

**Review by Sabirah42 — CHANGES_REQUESTED** [review]
At least one test needed for controllers - cannot approve without as we need to check testing works with cloud Mongo (we haven't been able to make it work as yet either)

**Review by tlchambers — COMMENTED** [review]
I know I have been banging on about testings this morning and on a risk of sounding like a broken record, could we bare in mind we need to create a spec folder containing model and controller folders for testings please

File: backend/controllers/listingsController.js (hunk 1 above)
Comment (SyntheticDBX) [reply backend/controllers/listingsController.js]: I've refactored it to a map and have pushed to the branch. Can see changes on the branch.

**Pull request closed without merging**